src/libxtp/molden.cc

```diff
@@ -0,0 +1,264 @@
+#include <sstream>
+
+#include "votca/xtp/basisset.h"
+#include "votca/xtp/molden.h"
+#include <boost/algorithm/string.hpp>
+
+namespace votca {
+namespace xtp {
+
+void Molden::writeAtoms(const Orbitals& orbitals,
+                        std::ofstream& outFile) const {
+  for (const auto& atom : orbitals.QMAtoms()) {
+    const Eigen::Vector3d& pos = atom.getPos();
+    outFile << boost::format("%-4s %5d %5d %22.12e %22.10e %22.10e\n") %
+                   atom.getElement() % (atom.getId() + 1) %
+                   atom.getTotalNuccharge() % pos[0] % pos[1] % pos[2];
+  }
+}
+
+void Molden::writeMOs(const Orbitals& orbitals, std::ofstream& outFile) const {
+  Eigen::VectorXd energies = orbitals.MOs().eigenvalues();
+  OrbReorder reorder(_reorderList, _multipliers);
+  Eigen::MatrixXd moCoefficients = orbitals.MOs().eigenvectors();
+  reorder.reorderOrbitals(moCoefficients, _basis);
+
+  for (Index i = 0; i < orbitals.getBasisSetSize(); i++) {  // over columns
+    outFile << "Sym= \n";
+    outFile << boost::format("Ene= %-20.12e\n") % energies[i];
+    outFile << "Spin= Alpha\n";
+    outFile << boost::format("Occup= %-5.2f\n") %
+                   (2 * (i < orbitals.getLumo()));
+    for (Index j = 0; j < orbitals.getBasisSetSize(); j++) {
+      outFile << boost::format("%5d %22.12e\n") % (j + 1) %
+                     moCoefficients(j, i);
+    }
+  }
+}
+
+void Molden::writeBasisSet(const Orbitals& orbitals,
+                           std::ofstream& outFile) const {
+
+  for (const auto& atom : orbitals.QMAtoms()) {
+    const Element& element = _bs.getElement(atom.getElement());
+    // The 0 in the format string of the next line is meaningless it
+    // is included for backwards compatibility of molden files
+    outFile << boost::format("%4d 0 \n") % (atom.getId() + 1);
+    for (const Shell& shell : element) {
+      // The 1.0 at the end of the next line is meaningless it
+      // is included for backwards compatibility of molden files
+      outFile << boost::format("%-3s %4d %3.1f \n") %
+                     boost::to_lower_copy(EnumToString(shell.getL())) %
+                     shell.getSize() % 1.0;
+      for (const GaussianPrimitive& gaussian : shell) {
+        outFile << boost::format("%22.10e %22.10e\n") % gaussian.decay() %
+                       gaussian.contraction();
+      }
+    }
+    outFile << " \n";
+  }
+}
+
+void Molden::WriteFile(const std::string& filename, const Orbitals& orbitals) {
+  if (orbitals.hasDFTbasisName()) {
+    _bs.Load(orbitals.getDFTbasisName());
+    _basis.Fill(_bs, orbitals.QMAtoms());
```

Comment:
do you actually need basisset and aobasis?

Reply:
yes, the basisset for writing the basisset to molden and the aobasis to reorder the MO coefficients.

Reply:
why not just write the aobasis to molden?